### PR TITLE
Source command

### PR DESCRIPTION
**Pull request by chrisseaton**
Tells you why a node is in the graph - the stack trace in the abstract interpreter that builds the graph at the point when the node was created.

```
% seafoam examples/fib-ruby.bgv:8:2544 source
  java.lang.Math#addExact
  org.truffleruby.core.numeric.IntegerNodes$AddNode#add
  org.truffleruby.core.numeric.IntegerNodesFactory$AddNodeFactory$AddNodeGen#executeAdd
  org.truffleruby.core.inlined.InlinedAddNode#intAdd
  org.truffleruby.core.inlined.InlinedAddNodeGen#execute
  org.truffleruby.language.control.IfElseNode#execute
  org.truffleruby.language.control.SequenceNode#execute
  org.truffleruby.language.arguments.CheckArityNode#execute
  org.truffleruby.language.control.SequenceNode#execute
  org.truffleruby.language.methods.CatchForMethodNode#execute
  org.truffleruby.language.methods.ExceptionTranslatingNode#execute
  org.truffleruby.language.RubyRootNode#execute
  org.graalvm.compiler.truffle.runtime.OptimizedCallTarget#executeRootNode
  org.graalvm.compiler.truffle.runtime.OptimizedCallTarget#profiledPERoot
```

### CODE REVIEW

File: spec/seafoam/command_spec.rb [hunk 1]
```diff
@@ -127,7 +128,7 @@
 
   describe '#props' do
     it 'prints properties for a file' do
-      @commands.send :props, @fib_java.to_s
+      @commands.send :props, @fib_java
```
Comment (tomstuart) [inline spec/seafoam/command_spec.rb:131]: Is this change intentional? It doesn’t look related.

File: lib/seafoam/commands.rb [hunk 2]
```diff
@@ -20,6 +20,7 @@ def seafoam(*args)
         @out.puts '        file.bgv[:graph][:node[-edge]] search term...'
         @out.puts '        file.bgv[:graph][:node[-edge]] edges'
         @out.puts '        file.bgv[:graph][:node[-edge]] props'
+        @out.puts '        file.bgv[:graph]:node source'
```
Comment (tomstuart) [inline lib/seafoam/commands.rb:23]: Is the graph index really optional? `#source` looks like it requires it.

File: lib/seafoam/graal/source.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+module Seafoam
+  module Graal
+    # Routines for understanding source positions in Graal.
+    module Source
+      def self.render(source_position)
+        lines = []
+        caller = source_position
+        while caller
+          method = caller[:method]
+          lines.push '  ' + render_method(method)
```
Comment (tomstuart) [inline lib/seafoam/graal/source.rb:10]: Is the indent important? It makes `#source` a little harder to test (the `gsub(/^\s*/, '')` business) and could be mildly inconvenient if you wanted to pipe the output of `seafoam source` elsewhere.

**Review by tomstuart — APPROVED** [review]
Code looks good and works for me! A few clarifying questions but nothing that should block merging. 👍🏻

File: spec/seafoam/command_spec.rb (hunk 1 above)
Comment (chrisseaton) [reply spec/seafoam/command_spec.rb:131]: Unrelated tidy-up - `@fib_java` is already a `String`!

File: lib/seafoam/commands.rb (hunk 2 above)
Comment (chrisseaton) [reply lib/seafoam/commands.rb:23]: Ah, not it's not - thanks!

File: lib/seafoam/graal/source.rb (hunk 3 above)
Comment (chrisseaton) [reply lib/seafoam/graal/source.rb]: Fixed.
